### PR TITLE
scheduler: unify modification of s.bounds and s.unbounds

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
reduce a TODO

### What is changed and how it works?
`s.bounds` and `s.unbounds` should be mostly modified inside `updateStatusToBound` and `updateStatusToUnbound`

### Check List 

Tests 

 - pass original test

Code changes

Side effects

Related changes

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

**Review by lichunzhu — APPROVED** [review]

File: dm/dm/master/scheduler/scheduler.go [hunk 1]
```diff
@@ -2094,33 +2049,37 @@ func (s *Scheduler) deleteWorker(name string) {
...
+// updateStatusToUnbound updates the in-memory status for unbound, including:
 // - update the stage of worker to `Free` or `Relay`.
 // - remove the bound relationship in the scheduler.
+// - record the unbound relationship in the scheduler.
 // this func is called after the bound relationship removed from etcd.
-func (s *Scheduler) updateStatusForUnbound(source string) {
+func (s *Scheduler) updateStatusToUnbound(source string) {
 	w, ok := s.bounds[source]
 	if !ok {
+		s.unbounds[source] = struct{}{}
```
Comment (glorv) [inline dm/dm/master/scheduler/scheduler.go:2075]: We can move this line before this if since all branches need to do this.

**Review by glorv — APPROVED** [review]

**Review by glorv — CHANGES_REQUESTED** [review]
rest LGTM

File: dm/dm/master/scheduler/scheduler.go [hunk 2]
```diff
@@ -314,16 +314,11 @@ func (s *Scheduler) AddSourceCfg(cfg *config.SourceConfig) error {
...
 	// 4. try to bound it to a Free worker.
-	bounded, err := s.tryBoundForSource(cfg.SourceID)
-	if err != nil {
-		return err
-	} else if !bounded {
-		// 5. record the source as unbounded.
-		s.unbounds[cfg.SourceID] = struct{}{}
-	}
-	return nil
+	_, err = s.tryBoundForSource(cfg.SourceID)
```
[the remark this answers is not in the document]
Comment (lance6716) [reply dm/dm/master/scheduler/scheduler.go:320]: some TODO may need this `ok` value, to let caller give some information to user.

like https://github.com/pingcap/ticdc/blob/e052c6aa1878130fccc296be861a4e8201cfcaac/dm/dm/master/scheduler/scheduler.go#L1957

File: dm/dm/master/scheduler/scheduler.go (hunk 1 above)
Comment (lance6716) [reply dm/dm/master/scheduler/scheduler.go]: https://github.com/pingcap/ticdc/pull/3257/commits/571f48dd2ceb04924b00bf37be3b48c21b2ba02e

PTAL

**Review by glorv — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]

**Pull request merged**